DeleteObjects API times out for deleting object versions

<!--- Provide a general summary of the issue in the Title above -->

When running below `minio-java` test, server fails to return response and make the test timing out.

```java
import io.minio.*;
import io.minio.messages.*;
import java.io.*;
import java.util.*;

public class Check {
  public static void main(String[] args) throws Exception {
    MinioClient client = new MinioClient(args[0], args[1], args[2]);

    client.makeBucket(MakeBucketArgs.builder().bucket("bala-test-bucket2").build());
    client.enableVersioning(EnableVersioningArgs.builder().bucket("bala-test-bucket2").build());

    ObjectWriteResponse v1 =
        client.putObject(
            PutObjectArgs.builder().bucket("bala-test-bucket2").object("my-object").stream(
                    new ByteArrayInputStream(new byte[] {'a'}), 1, -1)
                .build());

    ObjectWriteResponse v2 =
        client.putObject(
            PutObjectArgs.builder().bucket("bala-test-bucket2").object("my-object").stream(
                    new ByteArrayInputStream(new byte[] {'b'}), 1, -1)
                .build());

    List<DeleteObject> objects = new LinkedList<>();
    objects.add(new DeleteObject("my-object", v1.versionId()));
    objects.add(new DeleteObject("my-object", v2.versionId()));

    System.out.println(
        "Removing my-object and it's versions [" + v1.versionId() + ", " + v2.versionId() + "]");

    for (Result<DeleteError> result :
        client.removeObjects(
            RemoveObjectsArgs.builder().bucket("bala-test-bucket2").objects(objects).build())) {
      DeleteError error = result.get();
      System.out.println("Error in deleting object " + error.objectName() + "; " + error.message());
    }

    client.removeBucket(RemoveBucketArgs.builder().bucket("bala-test-bucket2").build());
  }
}
```
```java
Removing my-object and it's versions [5149ee3a-0388-4dd6-887c-3cf669075e02, 2df77c34-2ab9-4bad-b802-92b76c094ba5]
Exception in thread "main" java.net.SocketTimeoutException: timeout
	at okio.Okio$4.newTimeoutException(Okio.java:232)
	at okio.AsyncTimeout.exit(AsyncTimeout.java:286)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:241)
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:358)
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230)
	at okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.java:242)
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:213)
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115)
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142)
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117)
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229)
	at okhttp3.RealCall.execute(RealCall.java:81)
	at io.minio.MinioClient.execute(MinioClient.java:1258)
	at io.minio.MinioClient.execute(MinioClient.java:1406)
	at io.minio.MinioClient.executePost(MinioClient.java:1574)
	at io.minio.MinioClient.deleteObjects(MinioClient.java:7629)
	at io.minio.MinioClient$1$1.populate(MinioClient.java:3320)
	at io.minio.MinioClient$1$1.hasNext(MinioClient.java:3352)
	at Check.main(Check.java:33)
Caused by: java.net.SocketException: Socket closed
	at java.net.SocketInputStream.read(SocketInputStream.java:204)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at okio.Okio$2.read(Okio.java:140)
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:237)
	... 27 more
```

The same test works fine with AWS S3.

## Expected Behavior
<!--- If you're describing a bug, tell us what should happen -->
<!--- If you're suggesting a change/improvement, tell us how it should work -->

## Current Behavior
<!--- If describing a bug, tell us what happens instead of the expected behavior -->
<!--- If suggesting a change/improvement, explain the difference from current behavior -->

## Possible Solution
<!--- Not obligatory, but suggest a fix/reason for the bug, -->
<!--- or ideas how to implement the addition or change -->

## Steps to Reproduce (for bugs)
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
1.
2.
3.
4.

## Context
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context helps us come up with a solution that is most useful in the real world -->

## Regression
<!-- Is this issue a regression? (Yes / No) -->
<!-- If Yes, optionally please include minio version or commit id or PR# that caused this regression, if you have these details. -->

## Your Environment
<!--- Include as many relevant details about the environment you experienced the bug in -->
* Version used (`minio version`): Development build using commit id `72743d15902717bbf656e4b7159c5e894b8c5312`
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version: `rm -fr ~/tmp/ed{1..4}; MINIO_ACCESS_KEY=minio MINIO_SECRET_KEY=minio123 ./minio server ~/tmp/ed{1..4}`
* Operating System and version (`uname -a`):
* Link to your project:

--- a/cmd/erasure-zones.go
+++ b/cmd/erasure-zones.go
@@ -26,6 +26,7 @@ import (
 	"sync"
 	"time"
 
+	"github.com/minio/minio-go/v6/pkg/set"
 	"github.com/minio/minio-go/v6/pkg/tags"
 	"github.com/minio/minio/cmd/config/storageclass"
 	xhttp "github.com/minio/minio/cmd/http"
@@ -542,14 +543,14 @@ func (z *erasureZones) DeleteObject(ctx context.Context, bucket string, object s
 func (z *erasureZones) DeleteObjects(ctx context.Context, bucket string, objects []ObjectToDelete, opts ObjectOptions) ([]DeletedObject, []error) {
 	derrs := make([]error, len(objects))
 	dobjects := make([]DeletedObject, len(objects))
-	objNames := make([]string, len(objects))
+	objSets := set.NewStringSet()
 	for i := range derrs {
 		derrs[i] = checkDelObjArgs(ctx, bucket, objects[i].ObjectName)
-		objNames[i] = objects[i].ObjectName
+		objSets.Add(objects[i].ObjectName)
 	}
 
 	// Acquire a bulk write lock across 'objects'
-	multiDeleteLock := z.NewNSLock(ctx, bucket, objNames...)
+	multiDeleteLock := z.NewNSLock(ctx, bucket, objSets.ToSlice()...)
 	if err := multiDeleteLock.GetLock(globalOperationTimeout); err != nil {
 		for i := range derrs {
 			derrs[i] = err

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -1181,7 +1181,11 @@ func (fs *FSObjects) DeleteObjects(ctx context.Context, bucket string, objects [
 	dobjects := make([]DeletedObject, len(objects))
 	for idx, object := range objects {
 		if object.VersionID != "" {
-			errs[idx] = NotImplemented{}
+			errs[idx] = VersionNotFound{
+				Bucket:    bucket,
+				Object:    object.ObjectName,
+				VersionID: object.VersionID,
+			}
 			continue
 		}
 		_, errs[idx] = fs.DeleteObject(ctx, bucket, object.ObjectName, opts)

